aa_stripe/models.py

```diff
@@ -214,6 +214,13 @@ def save(self, force_retrieve=False, *args, **kwargs):
                     coupon.metadata = self.metadata
                     coupon.save()
 
+                if force_retrieve:
+                    # make sure we are not creating a duplicate
+                    coupon_qs = StripeCoupon.objects.filter(coupon_id=self.coupon_id)
+                    if coupon_qs.filter(created=timestamp_to_timezone_aware_date(coupon["created"])).exists():
+                        raise StripeCouponAlreadyExists
+
+                    coupon_qs.update(is_deleted=True)  # all old coupons should be deleted
```

Comment:
Would not plan delete be better here? simple, and triggering signals.

Reply:
StripeCoupon.delete() will delete another save(), but having post save signal is good point.